### PR TITLE
cmd/utils, miner: A/B testing JIT VM. Disabled for miners

### PR DESCRIPTION
**Pull request by obscuren**
This PR introduces a 10% probability that you'll run the client with the
JIT enabled testing the new client and helps us potentially catch
errors when reported.

This feature is **disabled** for miners (disabling the JIT completely).
The JIT can however be force for miners if they enable both `--jitvm` and
`--forcejit`.

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Thu Mar 24 12:16:34 UTC 2016_

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -659,6 +661,16 @@ func MakeSystemNode(name, version string, extra []byte, ctx *cli.Context) *node.
 	// Configure the Ethereum service
 	accman := MakeAccountManager(ctx)
 
+	// initialise new random number generator
+	rand := rand.New(rand.NewSource(time.Now().UnixNano()))
```
Comment (fjl) [inline cmd/utils/flags.go]: Maybe also seed with os.GetPid()

Comment (obscuren) [inline cmd/utils/flags.go]: I don't think it's really that important. Why do you think we'd need additional entropy?

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -659,6 +661,16 @@ func MakeSystemNode(name, version string, extra []byte, ctx *cli.Context) *node.
...
+	// get enabled jit flag
+	jitEnabled := ctx.GlobalBool(VMEnableJitFlag.Name)
+	// if the jit is not enabled enable it for 10 pct of the people
+	if !jitEnabled && rand.Float64() < 0.1 {
+		jitEnabled = true
+		glog.V(logger.Info).Infoln("You're one of the lucky few that will try out the JIT VM (random). If you get a consensus failure please be so kind to report this incident with the block hash that failed. You can fix this issue by restarting")
```
Comment (fjl) [inline cmd/utils/flags.go]: "... You can fix this issue" sounds weird. Maybe it should say "you can switch to the regular VM using --flag-xxx".

Please break the text onto multiple lines.

File: cmd/utils/flags.go (hunk 1 above)
Comment (fjl) [inline cmd/utils/flags.go]: The only source of entropy here is time. Whether you get the JitVM or not depends entirely on the time when you start geth. I don't know whether that's an issue or not, just feels weird.

**Comment by fjl** [comment]
:+1:

**Pull request merged**